docs/polkadot/implementors-guide/guide.md

```diff
@@ -0,0 +1,544 @@
+  asynchronously {
+    let pov = (fetch pov block).await
+
+    // dispatched to sub-process (OS process) pool.
+    let valid = validate_candidate(candidate, validation function, parachain head, pov).await;
+    if valid {
+      // make PoV available for later distribution.
+      // sign and dispatch `valid` statement to network if we have not seconded the given candidate.
+    } else {
+      // sign and dispatch `invalid` statement to network.
+    }
+  }
+}
+```
+
+*fetch pov block*
+
+Create a `(sender, receiver)` pair.
+Dispatch a `PovFetchProcessMessage(relay_parent, candidate_hash, sender)` and listen on the receiver for a response.
+
+*on receiving CandidateBackingProcessMessage*
+* If the message is a `CandidateBackingProcessMessage::RegisterBackingWatcher`, register the watcher and trigger it each time a new candidate is backed. Also trigger it once initially if there are any backed candidates at the time of receipt.
+* If the message is a `CandidateBackingProcessMessage::Second`, sign and dispatch a `Seconded` statement only if we have not seconded any other candidate and have not signed a `Valid` statement for the requested candidate. Signing both a `Seconded` and `Valid` message is a double-voting misbehavior with a heavy penalty, and this could occur if another validator has seconded the same candidate and we've received their message before the internal seconding request.
+
+(TODO: send statements to Statement Distribution Process, handle shutdown signal from parent process)
+
+---
+
+### Candidate Proposal Process
+
+[TODO: get candidate from collator, feed to candidate backing process. ]
+
+### Secondary Checking
+
+[TODO]
+
+### Validator Assignment Process
+
+[TODO]
+
+### Collation Distribution Process
+
+[TODO]
+
+### Availability Distribution Process
+
+[TODO]
+
+----
+
+## Data Structures and Types
+
+[TODO]
+* CandidateReceipt
+* CandidateCommitments
+* AbridgedCandidateReceipt
+* GlobalValidationSchedule
+* LocalValidationData
+
+#### Block Import Event
+```rust
+/// Indicates that a new block has been added to the blockchain.
+struct BlockImportEvent {
+  /// The block header-hash.
+  hash: Hash,
+  /// The header itself.
+  header: Header,
+  /// Whether this block is considered the head of the best chain according to the 
+  /// event emitter's fork-choice rule.
+  new_best: bool,
+}
+```
+
+#### Block Finalization Event
+```rust
+/// Indicates that a new block has been finalized.
+struct BlockFinalizationEvent {
+  /// The block header-hash.
+  hash: Hash,
+  /// The header of the finalized block.
+  header: Header,
+}
+```
+
+#### Statement Type
+```rust
+/// A statement about the validity of a parachain candidate.
+enum Statement {
+  /// A statement about a new candidate.
+  Seconded(CandidateReceipt),
+  /// A statement about the validity of a candidate, based on candidate's hash.
+  Valid(Hash),
+  /// A statement about the invalidity of a candidate.
+  Invalid(Hash),
+}
+```
+
+#### Signed Statement Type
+
```

Comment:
`This is a signed statement.` does not add much value to the block :)

Not sure if you want to go into the purpose in this section.